src/Rest/Routes/FormSubmitCaptchaRoute.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * The class register route for public form submiting endpoint - Captcha
+ *
+ * @package EightshiftForms\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Rest\Routes;
+
+use EightshiftForms\Helpers\Components;
+use EightshiftForms\Hooks\Variables;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftForms\Labels\LabelsInterface;
+use EightshiftForms\Validation\SettingsCaptcha;
+
+/**
+ * Class FormSubmitCaptchaRoute
+ */
+class FormSubmitCaptchaRoute extends AbstractBaseRoute
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Instance variable of LabelsInterface data.
+	 *
+	 * @var LabelsInterface
+	 */
+	protected $labels;
+
+	/**
+	 * Create a new instance that injects classes
+	 *
+	 * @param LabelsInterface $labels Inject LabelsInterface which holds labels data.
+	 */
+	public function __construct(LabelsInterface $labels)
+	{
+		$this->labels = $labels;
+	}
+
+	/**
+	 * Get the base url of the route
+	 *
+	 * @return string The base URL for route you are adding.
+	 */
+	protected function getRouteName(): string
+	{
+		return '/form-submit-captcha';
+	}
+
+	/**
+	 * Get callback arguments array
+	 *
+	 * @return array<string, mixed> Either an array of options for the endpoint, or an array of arrays for multiple methods.
+	 */
+	protected function getCallbackArguments(): array
+	{
+		return [
+			'methods' => $this->getMethods(),
+			'callback' => [$this, 'routeCallback'],
+			'permission_callback' => [$this, 'permissionCallback'],
+		];
+	}
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param \WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @return \WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(\WP_REST_Request $request)
+	{
+		$params = json_decode($request->get_body(), true);
```

Comment:
If the request body is not valid JSON, this line will assign `$params = false`, which will in turn cause lines 83 and 84 to throw Undefined index warnings.

I'd suggest doing one of these two things:

```suggestion
		$params = json_decode($request->get_body(), true);
		if (!$params) { 
				return;
		}
```

or (in PHP 7.3+)

```suggestion
		try {
				$params = json_decode($request->get_body(), true, null, JSON_THROW_ON_ERROR);
		}
		catch (\Throwable $t) {
						return \rest_ensure_response([
								'status' => 'error',
								'code' => 400,
								'message' => $this->labels->getLabel('mailerSuccessNoSend', $formId),
							]);
		}
```

(Note that this code was written in the Github comment field, so please lint it if commited 😄 )

Reply:
nice catch tnx. Updated